sdk/appcenter-storage/src/test/java/com/microsoft/appcenter/storage/StorageTest.java

```diff
@@ -476,6 +457,18 @@ public void readEndToEnd() throws Exception {
         TestDocument testDocument = testCosmosDocument.getDocument();
         assertNotNull(testDocument);
         assertEquals(TEST_FIELD_VALUE, testDocument.test);
+
+        /* The test passes individually, but not in a suit. Need to figure out which mocks do not get reset. */
+        when(mNetworkStateHelper.isNetworkConnected()).thenReturn(false);
+        doc = Storage.read(PARTITION, DOCUMENT_ID, TestDocument.class);
+        verifyNoMoreInteractions(mHttpClient);
+        verify(mDocumentCache).read(
+                eq(PARTITION),
+                eq(DOCUMENT_ID),
+                eq(testDocument.getClass()),
+                any(ReadOptions.class));
+
+        // TODO add asserts for doc
```

Comment:
There's no way to assert on the doc, turns out, because the underlying call goes to Sqlite. And since this is a regular unit test, we can't actually get the return value. So I just made sure that the call goes through with the right parameters and we'll rely on `DocumentCache` tests to make sure that what's beneath works as expected.